### PR TITLE
Limit number of threads used by OMP in circleci.

### PR DESCRIPTION
**Pull request by rossbar**
May speed up any long-running examples that rely on numpy/scipy linalg.

See how the sphinx-gallery build goes in circleci, apply to other CI services if it has an effect.

### CODE REVIEW

**Comment by jarrodmillman** [comment]
This makes a big difference for at least `examples/drawing/plot_eigenvalues.py`.  Before it took between 20 to 150 seconds.  Now it takes around 0.80 sec!

**Comment by rossbar** [comment]
Yeah I'm not sure why this is, but openblas threading does not seem to work well with circleci. This doesn't seem to affect the GH workflows though (e.g. see the sphinx-gallery timings for the doc deploy workflow), so I will limit the changes to the circleci config.

**Comment by jarrodmillman** [comment]
How do you see the  doc deploy workflow?  I thought that was only run on main.  Have you tried it on the coverage report workflow?  Just curious.

**Comment by rossbar** [comment]
> How do you see the doc deploy workflow?

If you go to the main page on GH, you can click on the (hopefully) green checkmark at the top of the file explorer to see the jobs that run on `main` post-merge

> Have you tried it on the coverage report workflow? Just curious.

Not for the coverage job, but I did try it on the test workflow for the `extras` jobs (i.e. the ones with numpy, scipy, etc.) There was a slight difference in the test suite run time (115sec vs 130 sec), but it's not clear that the limiting had any real affect. Also, I took the deploy job not being affected as evidence that this bad blas/server interaction was limited to circleci.

**Comment by jarrodmillman** [comment]
I am still confused.  This hasn't been merged to `main`, so how could you see the results of the PR for something that is only run after the PR is merged to `main`.  For instance, if I look at
https://github.com/networkx/networkx/actions/workflows/deploy-docs.yml
I don't see any runs that contain `export OMP_NUM_THREADS=1`.

**Comment by rossbar** [comment]
Ah sorry, I never explained what I was looking at...

I was looking at the [runtime for the `plot_eigenvalues.py` gallery example (not sure if this link will work)](https://github.com/networkx/networkx/runs/2641432346) that was running slowly on circleci. In the GH doc deploy workflow from the previous merge (which, as you point out, doesn't have any special OMP tweaking) it was ~1.1 sec - pretty much in line with the circleci runtime after the OMP tweak.

**Comment by rossbar** [comment]
I'll try experimenting with the other CI workflows a bit more this evening.

**Comment by jarrodmillman** [comment]
Ah!  If it doesn't have the problem, it doesn't need the fix.  That is some sound reasoning.

**Review by jarrodmillman — APPROVED** [review]

**Comment by jarrodmillman** [comment]
@rossbar Is this ready to merge (or did you want to test anything else first)?  This LGTM.

**Comment by rossbar** [comment]
Nope this is good to go, I'll push off any other CI experimentation to other PRs 👍

**Pull request merged**
